src/FSharp.AWS.DynamoDB/TableContext.fs

```diff
@@ -846,110 +1021,103 @@ type TableContext<'TRecord> internal
-            | _ -> None
-
-        let rec verify lastExn retries = async {
-            match lastExn with
-            | Some e when retries = 0 -> do! Async.Raise e
-            | _ -> ()
+    member _.VerifyTableAsync() : Async<unit> =
+        Provisioning.validateOnly (client, tableName, template)
 
-            let! ct = Async.CancellationToken
-            let! response =
-                client.DescribeTableAsync(tableName, ct)
-                |> Async.AwaitTaskCorrect
-                |> Async.Catch
-
-            match response with
-            | Choice1Of2 td ->
-                if td.Table.TableStatus <> TableStatus.ACTIVE then
-                    do! Async.Sleep 2000
-                    // wait indefinitely if table is in transition state
-                    return! verify None retries
-                else
-
-                let existingSchema = TableKeySchemata.OfTableDescription td.Table
-                if existingSchema <> template.Info.Schemata then
-                    sprintf "table '%s' exists with key schema %A, which is incompatible with record '%O'."
-                        tableName existingSchema typeof<'TRecord>
-                    |> invalidOp
-
-            | Choice2Of2 (:? ResourceNotFoundException) when createIfNotExists ->
-                let provisionedThroughput =
-                    match provisionedThroughput with
-                    | None -> ProvisionedThroughput(10L,10L)
-                    | Some pt -> pt
-
-                let ctr = template.Info.Schemata.CreateCreateTableRequest (tableName, provisionedThroughput)
-                let! ct = Async.CancellationToken
-                let! response =
-                    client.CreateTableAsync(ctr, ct)
-                    |> Async.AwaitTaskCorrect
-                    |> Async.Catch
-
-                match response with
-                | Choice1Of2 _ -> return! verify None retries
-                | Choice2Of2 (Conflict as e) ->
-                    do! Async.Sleep 2000
-                    return! verify (Some e) (retries - 1)
-
-                | Choice2Of2 e -> do! Async.Raise e
-
-            | Choice2Of2 (Conflict as e) ->
-                do! Async.Sleep 2000
-                return! verify (Some e) (retries - 1)
-
-            | Choice2Of2 e -> do! Async.Raise e
-        }
+    /// <summary>
+    /// Adjusts the Table's configuration via <c>UpdateTable</c> if the <c>throughput</c> or <c>streaming</c> are not as specified.<br/>
+    /// NOTE: The underlying API can throw if a change is currently in progress; see the DynamoDB <a href="https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_UpdateTable.html"><c>UpdateTable</c> API documentation</a>.<br/>
+    /// NOTE: Throws <c>InvalidOperationException</c> if the table is not yet <c>Active</c>. It is recommended to ensure the Table is prepared via <c>VerifyTableAsync</c> or <c>VerifyOrCreateTableAsync</c> to guard against the potential for this state.
```

Comment:
Ultimately I'd like to see a list of `<exception>` tags on the doc comments rather than pointing people at documentation, but don’t worry about this now, I'll go through and do an audit later on.

Reply:
I was trying to avoid translating e.g. the restrictions on number of reductions in a 24h period into a terser form - that just leaves you hostage to policy changes. This is why .NET needs checked exceptions ;)

If it's ok with you I'll leave it (the exception specs should be proper xmldoc syntax throughout too...)